### PR TITLE
CLOUDP-57045: CLI: Updating Atlas Database Users

### PR DESCRIPTION
**Pull request by andreaangiolillo**

### CODE REVIEW

File: internal/cli/atlas_dbusers_update.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	if opts.ProjectID() == "" {
+		return errMissingProjectID
+	}
+	var err error
+	opts.store, err = store.New()
+	return err
+}
+
+func (opts *atlasDBUsersUpdateOpts) Run() error {
+	current, err := opts.store.DatabaseUser(opts.orgID, opts.username)
```
Comment (gssbzn) [inline internal/cli/atlas_dbusers_update.go]: @andreaangiolillo this is your error, this should be `opts.ProjectID()`

Comment (gssbzn) [reply internal/cli/atlas_dbusers_update.go]: Also, de we really need to fetch the record first? is there any value to it?

Comment (andreaangiolillo) [reply internal/cli/atlas_dbusers_update.go]: I don’t understand your comment on the pull request. Does the command update an existing record right? so I need to get that record in order to append the new roles to it. what am I missing?

Comment (andreaangiolillo) [reply internal/cli/atlas_dbusers_update.go]: How do we know if the username used in the command is associated to a database user if we don't fetch the record first?

File: internal/convert/database_user.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import (
+	"strings"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+)
+
+const (
+	AdminDB = "admin"
+	RoleSep = "@"
```
Comment (gssbzn) [inline internal/convert/database_user.go]: could we make this private?

File: internal/convert/database_user.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+)
+
+func BuildRoles(r []string) []atlas.Role {
```
Comment (gssbzn) [inline internal/convert/database_user.go]: please add docs (`// BuildRoles ...`) to what this does

File: e2e/atlas_dbusers_test.go [hunk 4]
```diff
@@ -81,6 +81,33 @@ func TestAtlasDBUsers(t *testing.T) {
...
+			t.Fatalf("unexpected error: %v, resp: %v", err, string(resp))
+		}
+
+		user := mongodbatlas.DatabaseUser{}
+		err = json.Unmarshal(resp, &user)
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+
+		if user.Username != username {
```
Comment (gssbzn) [inline e2e/atlas_dbusers_test.go]: Thanks for adding an e2e test, could we add a check to see we actually changed the roles? I guess that's the only thing that would change from this API

File: internal/cli/atlas_dbusers_create.go [hunk 5]
```diff
@@ -16,7 +16,8 @@ package cli
 
 import (
 	"errors"
-	"strings"
+
+	"github.com/mongodb/mcli/internal/convert"
```
Comment (gssbzn) [inline internal/cli/atlas_dbusers_create.go]: [nit] sometimes importfmt misses the cases where there's an extra empty line, please delete the line and re run `make fmt`

File: internal/cli/atlas_dbusers_update.go [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+	"github.com/mongodb/mcli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type atlasDBUsersUpdateOpts struct {
+	*globalOpts
+	username string
+	password string
+	roles    []string
+	store    store.DatabaseUserStore
```
Comment (gssbzn) [inline internal/cli/atlas_dbusers_update.go]: could we reduce this surface to just be an `Updater`?

File: internal/cli/atlas_dbusers_update_test.go [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+// limitations under the License.
+
+package cli
+
+import (
+	"testing"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"github.com/mongodb/mcli/internal/fixtures"
+
```
Comment (gssbzn) [inline internal/cli/atlas_dbusers_update_test.go]: [nit] please delete the extra line

**Pull request merged**